docs/docsite/rst/lookup_guide.rst

```diff
@@ -0,0 +1,159 @@
+
+The way this relates to Ansible and this collection, is that we may have lookup plugins that either unintuitively perform writes (like ``vault_login``), or appear inappropriate to exist as lookups in the first place, like the planned ``vault_write`` lookup.
+
+The reason for this is that the we often consider these operations to be logical "read" operations, like performing a login, and want to use their results in other expressions.
+
+Something like ``vault_write`` does not always fit that description, because you could use it in a way that is clearly an explicit write, for example you could create a new policy with the lookup. But there are times it may be appropriate to use it in lookup semantics, like when "retrieving" (really creating) a new secret ID for an approle.
+
+When considering built-in support for auth methods, any auth method other than ``token`` or ``none`` makes every lookup, even ``vault_read``, into something that's changing state and performing a write within Vault. This actually applies to many modules too, even when using check mode.
+
+How to reason about when to use lookups
+---------------------------------------
+
+Because there is potential for writes in any lookup, it is very important to carefully consider when you are using a lookup vs. a module/other plugin. Check mode has no effect on lookups, so there is potential to perform many writes within your check mode run, but maybe sometimes you want that, for example if you're performing a ``vault_login`` via lookup to retrieve a token to use in your module calls, you may want that to still happen in check mode so that your module calls can properly check the things they need to.
+
+Some modules that are read focused, like the ``vault_read`` module, when used with auth other than ``token`` or ``none``, will still perform an internal login even in check mode, so this is still another consideration.
+
+Lookups and lazy templating
+---------------------------
+
+The capacity for lookups to perform writes or change state is exacerbated by Ansible's "lazy" templating, if not used carefully.
+
+Consider the following example:
+
+.. code-block:: yaml+jinja
+
+    - vars:
+        token: "{{ lookup('community.hashi_vault.vault_login', auth_method='userpass', username='user', password='pass') | community.hashi_vault.vault_login_token }}"
+        secret: "{{ lookup('community.hashI_vault.vault_read', 'secrets/data/my-secret', token=token) }}"
+        value_a: "{{ secret.data.data.a }}"
+        value_b: "{{ secret.data.data.b }}"
+      ansible.builtin.debug:
+        msg: "Secret value A is '{{ value_a }}' while value B is '{{ value_b }}'."
+
+Since templating is recursive and evaluated lazily, this will unfortunately *not* result in a single login, resuing the token to perform a single secret read, which is then used is dictionary lookups.
```

Comment:
```suggestion
Since templating is recursive and evaluated lazily, this will unfortunately *not* result in a single login, reusing the token to perform a single secret read, which is then used is dictionary lookups.
```